### ISSUE TITLE
StreamBuffer can use unlimied memory if the underlying stream is always ready

### ISSUE THREAD

**Issue by zl-gcharnock** [rwf2/multer-rs#62]
This is probably quite hard to encounter in the real world with a real network--I hit this while testing with artifical streams made from `futures::stream::repeat`. I was trying to write a unit test showing that if the network dropped I was properly cleaning up half-written files. Unfortunatly I struggled to get my unit test to produce a half-written file because the entire thing was being buffered in memory.

I think the problem is the following fuction in `buffer.rs`. If `self.stream` never returns pending, `self.buf.extend_from_slice(&data)` will keep being called and more and more data will be buffered up. Possibly breaking out of the loop if the buffer has reached some decent size might be a good idea? Not sure what "decent size" would be defined as here.

I fully understand if you don't agree that this is a problem, I appreciate that this is probably very much a edge case.

```rust 
    pub fn poll_stream(&mut self, cx: &mut Context<'_>) -> Result<(), crate::Error> {
        if self.eof {
            return Ok(());
        }

        loop {
            match self.stream.as_mut().poll_next(cx) {
                Poll::Ready(Some(Ok(data))) => {
                    self.stream_size_counter += data.len() as u64;

                    if self.stream_size_counter > self.whole_stream_size_limit {
                        return Err(crate::Error::StreamSizeExceeded {
                            limit: self.whole_stream_size_limit,
                        });
                    }

                    self.buf.extend_from_slice(&data)
                }
                Poll::Ready(Some(Err(err))) => return Err(err),
                Poll::Ready(None) => {
                    self.eof = true;
                    return Ok(());
                }
                Poll::Pending => return Ok(()),
            }
        }
    }
```

**Comment by SergioBenitez** [comment]
There's a limit check right above the line you're indicating as problematic. Have you set up [data limits](https://docs.rs/multer/latest/multer/#prevent-denial-of-service-dos-attack)? They exist to prevent exactly this.

**Comment by Erik1000** [comment]
@zl-gcharnock Thank you for your workaround. I just noticed this issue while using this on an ESP32 with very limited memory. Even tho I used chunks, the library loaded everything from the stream at once, leading to out of memory errors. Using your workaround helps, but I think this issue should stay open until either a note is added to the docs or it is fixed.

I think it should not read the file contents at all until `Field::chunks` is being called and then it should only ever read one item from the stream.
